internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1361,7 +1359,28 @@ func (r *ClusterDeploymentsReconciler) deleteClusterInstall(ctx context.Context,
 	return buildReply(err)
 }
 
-func (r *ClusterDeploymentsReconciler) UnbindAgents(ctx context.Context, log logrus.FieldLogger, clusterDeployment types.NamespacedName) error {
+func (r *ClusterDeploymentsReconciler) shouldDeleteAgentOnUnbind(ctx context.Context, agent aiv1beta1.Agent, clusterDeployment types.NamespacedName) bool {
+	infraEnvName, ok := agent.Labels[aiv1beta1.InfraEnvNameLabel]
```

Comment:
I don't see the agent controller setting this anywhere, but we do set it on create.

Should I also add something to the agent controller to ensure this is always set in this PR?

Reply:
Opened as https://issues.redhat.com/browse/MGMT-10969